com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -766,6 +766,8 @@ void ResetData()
         ///
         /// Your heuristic implementation can use any decision making logic you specify. Assign decision
         /// values to the float[] array, <paramref name="actionsOut"/>, passed to your function as a parameter.
+        /// The default actions passed to the `Heuristic()` method are the previous actions decided by the
```

Comment:
I think we should under-specify this. The actions could be changed in place by Agent.OnActionRecieved, but this makes it sound like we'd try to restore them (which isn't the case).

Reply:
Maybe something like 
> The same array will be reused between steps. It is up to the user to initialize the values on each call, for example by calling `Array.Clear(actionsOut, 0, actionsOut.Length);`.